torch/csrc/api/src/nn/module.cpp

```diff
@@ -7,6 +7,13 @@
 #include <unordered_map>
 
 namespace torch { namespace nn {
+
+Module::Module(const char* name) : name_(name) {}
```

Comment:
Are we sure we always want to require people from specifying a name for the module? Maybe we could have a default implementation of `name()` that uses `typeid(*this).name()` (possibly with a roundtrip through our demangling code)?